### DISCUSSION TITLE
[Search] [Feature Request] Web results in search

### DISCUSSION THREAD

**Discussion by CorruptionVR** [DustinBrett/daedalOS#390] [Feature Requests]
The way this would work is when there are no local results available, it would show previews in the search panel with an option to open the page or search in the browser.

**Comment by DustinBrett** [comment]
Ya that would be cool and is possible with an iframe and something similar to what I do in the Browser app. The issue I ran into was that the iframe can't resize to a mobile view and the Google results are too big. I could try a different search engine but at the moment I am ok to leave that feature out.

**Reply by CorruptionVR** [reply]
You.com and startpage.com have automatic mobile views.
Currently looking for others

**Reply by DustinBrett** [reply]
It needs to be a specific type of site with CORS allowed if it could be iframed on my site with results.

![image](https://github.com/DustinBrett/daedalOS/assets/16656318/164d723f-25bf-4f81-ab66-47d542655301)

![image](https://github.com/DustinBrett/daedalOS/assets/16656318/4d4ce7ac-1009-48e5-a87c-326c30c08e6a)

**Reply by DustinBrett** [reply]
Also automatic redirect/detect is based on user agent which also can't be controlled via the iframe. To get this to work it would also have to work in the [Browser](https://dustinbrett.com/?app=Browser) app.

**Reply by SuperMaxusa** [reply]
Maybe I could be wrong but how about to use DuckDuckGo API or Bing Web Search API for fetching results (if you talking about Search in taskbar)?

**Reply by DustinBrett** [reply]
Thanks for the suggestion but I only use API's which don't require keys and I keep everything client side. If there is a CORS accessible API on the front end which requires no API key, id consider that usage.

**Comment by DustinBrett** [comment]
> Maybe I could be wrong but how about to use DuckDuckGo API or Bing Web Search API for fetching results (if you talking about Search in taskbar)?

Actually maybe you are onto something as it seems to be open like I was hoping. I will dig further. Thanks for the suggestion.

![image](https://github.com/DustinBrett/daedalOS/assets/16656318/a0fff1fa-bb70-46b9-b57c-013c72c390ca)

As for Bing though, it seems to need a api key which I would like to avoid. But DuckDuckGo at the moment seems to work through CORS and without a key, so I could add that to Search maybe.

Although the DuckDuckGo API docs seem to be gone, the last existing one was https://web.archive.org/web/20220922150038/http://duckduckgo.com/api so I am unsure how much support it has.
